src/controllers/controllermanager.cpp

```diff
@@ -412,6 +412,8 @@ void ControllerManager::closeController(Controller* pController) {
             ConfigKey("[Controller]", sanitizeDeviceName(pController->getName())), 0);
 }
 
+// This needs to be called in a Qt::BlockingQueuedConnection so that the
```

Comment:
is there any way to enforce that / check that at runtime?

Reply:
Could we not doing it by adding a [thread affinity assertion](https://github.com/mixxxdj/mixxx/blob/633c84d9bdb4b3fc22febba2a93d4925ac72dd5e/src/util/thread_affinity.h#L13)?